### ISSUE TITLE
Dukascopy - WARNING or timeout causes missing data

### ISSUE THREAD

**Issue by halterc** [cuemacro/findatapy#43]
Hi. Thank you for this great package. However, I experienced a problem: When downloading Dukascopy forex tick data, if there is a:
`WARNING - Didn't download on 1 attempt`
or
`WARNING - Problem downloading.. https://www.dukascopy.com/datafeed/GBPJPY/2012/10/03/14h_ticks.bi5 HTTPConnectionPool(host='datafeed.dukascopy.com', port=80): Max retries exceeded with url: /datafeed/GBPJPY/2012/10/03/14h_ticks.bi5 (Caused by ConnectTimeoutError(<urllib3.connection.HTTPConnection object at 0x000001F49D7B09A0>, 'Connection to datafeed.dukascopy.com timed out. (connect timeout=10)')).. will try again 0 occasion`

it says "retry". But I noticed that it will cause missing data (in blocks of 1 hour, probably missing 1 h chunk file). It also causes to retry downloading seemingly in a random order also chunk files that were (I assume, as there was no warning) correctly downloaded (according to the spyder console, seems not in the logfile...).

Do you know what could be the culprit? Could it be the multiprocessing in combination with Windows? Or Spyder?
Data connection should be fine, I'm in a large institution. Firewall should not be the problem.

For your information, I'm on Windows 11 and using the following code:
```
SwimPool()
market = Market(market_data_generator=MarketDataGenerator())
DataConstants.market_thread_technique = "multiprocessing"
md_request = MarketDataRequest(start_date='01 Jan ' + str(year),
                                finish_date='31 Dec ' + str(year),
                                category='fx', fields=['bid', 'ask', 'bidv', 'askv'],
                                freq='tick', data_source='dukascopy', tickers=ticker)
df = market.fetch_market(md_request)
```

Thank you very much for your help.

**Comment by saeedamen** [comment]
Yes, you should probably remove the line changing the market thread technique. Downloading from DukasCopy seems to work ok from my machine (the main problem is if you increase the threads in the configuration, which can cause server side rejection - in the newest version of the code, I've reduced the number of threads for this reason). You can also try download a day at a time and sleep between calls.

**Comment by halterc** [comment]
Thank you for your answer. I tried without using the "thread" or "multiprocess" option. Unfortunately, the problem persisted (at least on my computers) that there will be missing data when the download is retried (I don't know why). I gave up and I sticked with Tickstory for downloading.

If I have already downloaded the data from Dukascopy, is there a possibility in findatapy to just unpack the .bi5 files for a ticker to a csv file, lets say as 1 min candles? That would be a really great functionality! :)
Thanks again for taking your time to create and maintain such a very useful library! 👍
